### PR TITLE
`Integrated code lifecycle`: Reduce service dependency in build job execution logic

### PR DESCRIPTION
**Pull request by laurenzfb**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

In the near future, we want to implement a new profile that only uses classes related to build job execution logic. Ideally, build agents are decoupled from most other Artemis functionality and only deal with the execution of build jobs. To achieve this decoupling, we need to do some refactoring.
The classes used for the execution of build jobs are pretty strongly coupled with other Artemis services. This includes database calls, submission result processing etc.. Ideally, we avoid these dependencies.

### Description

This PR is the first step of decoupling the build job execution logic. Before submitting the build job to the shared build job queue, we collect all necessary data that is needed to execute the build job and submit that information to the queue. Build agents listen to the queue and can execute build jobs given the build job information. 

In a second step (not this PR), we will decouple the submission result processing and user notification from the execution logic.

### Steps for Testing

**Only test on TS3, TS6, TS11, Local Staging or locally**
Prerequisites:
- 1 Instructor

1. Create a few programming exercises
2. Check if the template and solution results are as expected

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
  - Enhanced local continuous integration (CI) capabilities with new build configuration and job timing information.
  - Improved CI job management with streamlined parameters and methods for build job execution.

- **Bug Fixes**
  - Fixed an issue with CI job queue prioritization by using correct job timing info for comparison.

- **Refactor**
  - Refined local CI services and integration tests to use new directory constants and updated class dependencies.
  - Optimized CI job queue service with method refactoring and code reorganization.

- **Tests**
  - Updated integration tests to align with the latest changes in CI services and directory management.

- **Chores**
  - Cleaned up unused code and imports in local CI services to enhance maintainability.

### CODE REVIEW

**Review by konrad2002 — APPROVED** [review]
manually tested on legacy ts11, works as expected

**Review by LenaKahle — APPROVED** [review]
Approve, tested on legacy ts11.

File: src/main/java/de/tum/in/www1/artemis/config/Constants.java [hunk 1]
```diff
@@ -319,6 +319,16 @@ public final class Constants {
      */
     public static final int PUSH_NOTIFICATION_VERSION = 1;
 
+    /**
+     * The directory in the docker container in which the build script is executed
+     */
+    public static final String WORKING_DIRECTORY = "/var/tmp";
```
Comment (reschandreas) [inline src/main/java/de/tum/in/www1/artemis/config/Constants.java:325]: maybe prefix it with something related to localci

File: src/main/java/de/tum/in/www1/artemis/config/Constants.java [hunk 2]
```diff
@@ -319,6 +319,16 @@ public final class Constants {
...
+
+    /**
+     * The directory in the docker container in which the results can be found
+     */
+    public static final String RESULTS_DIRECTORY = "/results";
```
Comment (reschandreas) [inline src/main/java/de/tum/in/www1/artemis/config/Constants.java:330]: maybe prefix it with something related to localci

**Review by nata-81 — APPROVED** [review]
tested on ts1 works as expected.

**Review by milljoniaer — APPROVED** [review]
tested on ts6, worked as expected 🚀

**Review by reschandreas — APPROVED** [review]
tested locally, works as expected. code lgtm

**Review by mateusmm01 — APPROVED** [review]
Tested on ts3 works as expected! Code also lgtm

**Review by georgyia — APPROVED** [review]
manually tested on ts1, works as expected

**Review by MaximilianJG — APPROVED** [review]
Tested on ts3. Works as expected

**Pull request merged**